docs/architecture/api-20201231-preview.md

```diff
@@ -0,0 +1,124 @@
+# API 20201231-preview design
+
+For api `20201231-preview` bellow is the list of proposed changes and features.
+These would have to land into API and backend as a features.
+
+Feature work will need to be agreed and converted to working stories. All technical details bellow are overview only
+
+## API changes
+
+### .OpenShiftClusterProperties.SecurityProfile
+
+Add security profile for all security enhancements for the product.
+Proposal is to use new profile structure, so we do not scatter these options
+to corresponding objects. In example, `EncryptionAtHost` can be set on individual
+worker pools. But this creates possibility to configure some nodes with/without
+encryption.
```

Comment:
There are a couple of issues to balance here:
1) risk of setting some nodes without encryption (realistically, this is a risk regardless because end users can create new worker pools and get the setup wrong)
2) how do we report the case back when this happens?

I think both of the above may imply these settings need to go into the Master/WorkerProfiles.

Reply:
agreed. Was not in sane mind when did this